### DISCUSSION TITLE
Invalid input path with Reolink camera

### DISCUSSION THREAD

**Discussion by poperigby** [blakeblackshear/frigate#2750] [Ask A Question]
Hi. I'm getting this error when starting Frigate:
```
[2022-02-05 10:38:06] watchdog.reolink               ERROR   : FFMPEG process crashed unexpectedly for reolink.
[2022-02-05 10:38:06] watchdog.reolink               ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-05 10:38:06] watchdog.reolink               ERROR   : You may have invalid args defined for this camera.
[2022-02-05 10:38:06] ffmpeg.reolink.detect          ERROR   : http://192.168.1.3/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=admin&password=REDACTED: End of file
[2022-02-05 10:38:06] ffmpeg.reolink.record_rtmp     ERROR   : http://192.168.1.3/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=admin&password=REDACTED: End of file
[2022-02-05 10:38:06] watchdog.reolink               INFO    : Terminating the existing ffmpeg process...
[2022-02-05 10:38:06] watchdog.reolink               INFO    : Waiting for ffmpeg to exit gracefully...
```

Here's my configuration file:
```yaml
mqtt:
  host: mqtt
  port: 1883
  user: admin
  password: public
cameras:
  reolink:
    ffmpeg:
      # hwaccel_args:
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"
      inputs:
        - path: http://192.168.1.3/flv?port=1935&app=bcs&stream=channel0_main.bcs&user={FRIGATE_USER}&password={FRIGATE_PASSWORD}
          roles:
            - record
            - rtmp
        - path: http://192.168.1.3/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user={FRIGATE_USER}&password={FRIGATE_PASSWORD}
          roles:
            - detect
    detect:
      width: 640
      height: 480
      fps: 7
```

Docker Compose file:
```yaml
services:
  mqtt:
    image: emqx/emqx
    container_name: mqtt
    networks:
      - default
    volumes:
      - data:/opt/emqx/data
      - config:/opt/emqx/etc
      - log:/opt/emqx/log
  app:
    container_name: frigate
    hostname: frigate
    image: blakeblackshear/frigate:stable-amd64
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./config.yml:/config/config.yml:ro
      - /media/hdd0/media/frigate:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    environment:
      - FRIGATE_USER=admin
      - FRIGATE_PASSWORD=${FRIGATE_PASSWORD}
    ports:
      - 5000:5000
      - 1935:1935 # RTMP feeds
    networks:
      - default
      - proxy
    shm_size: "64mb" # update for your cameras based on calculation above
    privileged: true
    restart: unless-stopped
    depends_on:
      - mqtt
volumes:
  data:
  config:
  log:
networks:
  default:
  proxy:
    name: reverse_proxy
    external: true
```

**Comment by tkffaul** [comment]
I have reolink and everything looks ok to me, but why are you using the ext stream and down scaling it? Why not just use the sub stream for 640x480?

channel0_main <-- full res
channel0_ext <-- 896x672
channel0_sub <-- 640x480

I've had issues in the past trying to scale via CPU... I use the nvidia version with a LXC container and I can scale thru cuda on that... but I've just gone to using the ext as full res @ 5fps for my detect stream. This might not have helped but your config looks like mine and it works... so keep looking there must be something else going on.

**Reply by poperigby** [reply]
Thanks for the tip about using the sub stream.

> This might not have helped but your config looks like mine and it works... so keep looking there must be something else going on.

Thank you for at least trying.

**Comment by jmorris644** [comment]
On the input you are specifying rtmp and not using it? That could be one issue. If you want to use rtmp, this works for me.
```
- rtmpsrc location="rtmp://192.168.1.80/bcs/channel0_sub.bcs?channel=0&stream=1&user=user&password=password" do-timestamp=true timeout=20
- ```
-

**Reply by poperigby** [reply]
> On the input you are specifying rtmp and not using it?

I was just using [this](https://docs.frigate.video/configuration/camera_specific/#reolink-410520-possibly-others) configuration example.

> That could be one issue. If you want to use rtmp, this works for me.
> 
>`rtmpsrc location="rtmp://192.168.1.80/bcs/channel0_sub.bcs?channel=0&stream=1&user=user&password=password" do-timestamp=true timeout=20`

Where are you putting that? I don't see anything mentioning `rtmpsrc` in the documentation for the config file. And I've been meaning to ask, when you actually use that, are you replacing the `user` and `password` with your actual user and password?

**Comment by NickM-27** [comment]
This happens when passwords contain special characters or are too long. See https://github.com/blakeblackshear/frigate/pull/3719 as the docs will be updated to reflect this.
